Sprint_4

Nast8065 opened the pull request.

Nast8065 committed:
the added book has no genre

Nast8065 committed:
You can set the genre

Nast8065 committed:
inferring the genre of a book by its name

Nast8065 committed:
inferring the genre of a book by its name

Nast8065 committed:
the dictionary is displayed

Nast8065 committed:
we display the cartoons category

Nast8065 committed:
the book is added to favorites

Nast8065 committed:
the book is deleted from favorites

Nast8065 committed:
list of selected books

Nast8065 committed:
list of selected books

Nast8065 committed:
editing the README

Nast8065 committed:
editing the conftest

Nast8065 committed:
You can't add a new book with a title longer than 40 characters.

Nast8065 committed:
editing the README

Nast8065 committed:
editing the gitignore

ShamilovR commented on tests.py:
```diff
@@ -21,4 +21,89 @@ def test_add_new_book_add_two_books(self):
         assert len(collector.get_books_rating()) == 2
```
Нужно исправить: по заданию проверять данный тестовый метод не требуется

Nast8065 replied:
Я так поняла, что не нужно проверять что у любой из добавленных книг нет жанра. Я убрала эту проверку и добавила новую. Мне в этой программе не очень понятно что именно выделено для исправления. Как помечается, та проверка, которую нужно убрать.

ShamilovR replied:
Не совсем) 
Речь о методе get_books_rating, по заданию не требуется проверять данный метод. То есть использования данного метода не должно быть в решении, т.к. текущая реализация класса BooksCollector не предоставляет данного метода)

ShamilovR commented on tests.py:
```diff
@@ -21,4 +21,89 @@ def test_add_new_book_add_two_books(self):
+
+	#проверяем, что у любой из добавленных книг нет жанра
+    def test_add_new_book_one_book_has_no_genre(self):
+        collector = BooksCollector()
```
Можно лучше: здесь и далее: для работы с объектом BooksCollector следует использовать уже реализованную фикстуру collector

Nast8065 replied:
Готово!

Nast8065 committed:
editing the gitignore

Nast8065 committed:
editing the README

Nast8065 committed:
Update README.md

Nast8065 committed:
Update README.md

Updated README file

Nast8065 committed:
Update README.md

Nast8065 committed:
Delete README.md

Nast8065 committed:
Create README.md

Nast8065 committed:
Update README.md

Nast8065 committed:
Update README.md

new edited README file

Nast8065 committed:
Update tests.py

I removed the extra check and added a new one.

Nast8065 committed:
Update tests.py

I removed the unnecessary check and added a new one, prescribed the use of the collector fixture

Nast8065 committed:
Update tests.py

removed the verification get_books_rating

ShamilovR commented on tests.py:
```diff
@@ -4,21 +4,80 @@
+    def test_add_new_book_Identical_books_can_only_be_added_once(self, collector):
+        collector.add_new_book('Гордость и предубеждение и зомби')
+        collector.add_new_book('Гордость и предубеждение и зомби')
+        assert len(collector.get_books_rating()) == 1
```
Нужно исправить: по заданию проверять данный тестовый метод не требуется
Предыдущий [комментарий ](https://github.com/Nast8065/qa_python/pull/5#discussion_r2254200289)был исправлен не во всех местах проекта.

Nast8065 committed:
Update tests.py

we check that the added book has no genre

Nast8065 committed:
Update README.md

edited the file, added one check, and deleted the extra one
